packages/fxa-auth-server/scripts/prune-tokens.ts

```diff
@@ -245,39 +279,26 @@ Exit Codes:
           continue;
         }
 
-        // Pull session tokens from sql db.
-        const sessionTokens = await SessionToken.findByUid(uid);
-        const sqlSessionTokenIds = new Set(sessionTokens.map((x) => x.tokenId));
-
-        // Sanity check size of sqlSessionTokenIds and redisSessionTokenIds
-        if (maxSessions > 0 && sqlSessionTokenIds.size > maxSessions) {
-          log.warn(
-            `Found ${sqlSessionTokenIds.size}. Expected no more than ${maxSessions}.`
-          );
-        }
         if (redisSessionTokenIds.size > MAX_REDIS_SESSION_TOKENS) {
           log.warn(
             `found ${redisSessionTokenIds.size}. expected no more than ${MAX_REDIS_SESSION_TOKENS}.`
           );
         }
 
-        // Difference redisSessionTokenIds and sqlSessionTokenIds. This will result in a
-        // set of orphaned tokens.
-        const difference = new Set(redisSessionTokenIds);
-        for (const x of sqlSessionTokenIds) {
-          difference.delete(x);
-        }
+        log.info(`processing ${redisSessionTokenIds.size} for account ${uid}.`);
 
-        // Delete orphaned tokens from Redis
-        const orphanedTokenIds = [...difference];
-        if (orphanedTokenIds.length > 0) {
-          log.info(`pruning orphaned tokens from redis`, {
-            uid,
-            count: orphanedTokenIds.length,
-          });
-          await redis.pruneSessionTokens(uid, orphanedTokenIds);
-        } else {
-          log.info(`no orphaned tokens found in redis`);
+        // Iterate through current set of redis tokens, and remove any that
+        // don't exist in the sql database anymore.
+        for (const redisSessionTokenId of redisSessionTokenIds) {
+          const dbSessionTokens = await SessionToken.findByTokenId(
+            redisSessionTokenId
+          );
+          if (dbSessionTokens === null) {
+            log.info(
+              `pruning orphaned session in redis ${redisSessionTokenId}`
+            );
+            await redis.pruneSessionTokens(uid, [redisSessionTokenId]);
```

Comment:
Couldn't we build up a list in the loop and then call `pruneSessionTokens` just once?

Reply:
I was thinking about doing this. I've changed it as you suggest..